balancer-js/src/modules/pools/join/join.module.integration.spec.ts

```diff
@@ -0,0 +1,354 @@
+      {
+        forking: {
+          jsonRpcUrl,
+        },
+      },
+    ]);
+    await setupPools();
+    signerAddress = await signer.getAddress();
+    await setupTokenBalance(signerAddress, wETH, wETH_SLOT, initialBalance);
+    await setupTokenBalance(signerAddress, wBTC, wBTC_SLOT, initialBalance);
+    await approveTokens(tokensIn, [initialBalance, initialBalance], signer);
+  });
+
+  context('join transaction - join with encoded data', () => {
+    before(async function () {
+      this.timeout(20000);
+
+      amountsIn = [
+        parseFixed(wETH.balance, wETH.decimals).div('1000000').toString(),
+        parseFixed(wBTC.balance, wBTC.decimals).div('1000000').toString(),
+      ];
+
+      bptBalanceBefore = await tokenBalance(B_50WBTC_50WETH.address);
+      tokensBalanceBefore = [
+        await tokenBalance(wETH.address),
+        await tokenBalance(wBTC.address),
+      ];
+
+      const slippage = '100';
+      const { to, data, minBPTOut } = await balancer.pools.join.buildJoin(
+        signerAddress,
+        B_50WBTC_50WETH.id,
+        tokensInAddresses,
+        amountsIn,
+        slippage
+      );
+      const tx = { to, data };
+
+      bptMinBalanceIncrease = BigNumber.from(minBPTOut);
+      transactionReceipt = await (await signer.sendTransaction(tx)).wait();
+    });
+
+    it('should work', async () => {
+      expect(transactionReceipt.status).to.eql(1);
+    });
+
+    it('should increase BPT balance', async () => {
+      bptBalanceAfter = await tokenBalance(B_50WBTC_50WETH.address);
+
+      expect(
+        bptBalanceAfter.sub(bptBalanceBefore).toNumber()
+      ).to.greaterThanOrEqual(bptMinBalanceIncrease.toNumber());
+    });
+
+    it('should decrease tokens balance', async () => {
+      tokensBalanceAfter = [
+        await tokenBalance(wETH.address),
+        await tokenBalance(wBTC.address),
+      ];
+
+      for (let i = 0; i < tokensIn.length; i++) {
+        expect(
+          tokensBalanceBefore[i].sub(tokensBalanceAfter[i]).toString()
+        ).to.equal(amountsIn[i]);
+      }
+    });
+  });
+
+  context('join transaction - join with params', () => {
+    before(async function () {
+      this.timeout(20000);
+
+      amountsIn = [
+        parseFixed(wETH.balance, wETH.decimals).div('1000000').toString(),
+        parseFixed(wBTC.balance, wBTC.decimals).div('1000000').toString(),
+      ];
+
+      bptBalanceBefore = await tokenBalance(B_50WBTC_50WETH.address);
+      tokensBalanceBefore = [
+        await tokenBalance(wETH.address),
+        await tokenBalance(wBTC.address),
+      ];
+
+      const slippage = '100';
+      const { functionName, attributes, value, minBPTOut } =
+        await balancer.pools.join.buildJoin(
+          signerAddress,
+          B_50WBTC_50WETH.id,
+          tokensInAddresses,
+          amountsIn,
+          slippage
+        );
+      const transactionResponse = await balancer.contracts.vault
+        .connect(signer)
+        [functionName](
```

Comment:
Prettier is moving this to a new line while linter is raising an error. I'll search on how to fix this, but I'm happy with suggestions if anyone knows a solution.